.github/workflows/gh-pages.yaml

```diff
@@ -0,0 +1,37 @@
+name: GH-Pages
+on:
+  push:
+    branches:
+      - master
+    tags:
+      - "*"
+jobs:
+  build-and-deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Install Poetry
+        uses: snok/install-poetry@v1
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.9
+
+      - name: Install Dependencies
+        run: |
+          poetry install
+      - name: Build Documenations
+        run: |
+          poetry run python -m nox -s build-docs
+          touch doc/build/html/.nojekyll
+      - name: Deploy
+        uses: JamesIves/github-pages-deploy-action@v4.3.0
+        with:
+          branch: gh-pages
```

Comment:
Can this action do multiple versions? Why don't you use sphinx-github-pages generator

Reply:
wanted to have it as a to step thing, make sure the sphinx stuff is working, shown as expected then migrate.
I'll create a "follow" up ticket once this is merged.

Reply:
@tkilias I don't thinks so, just took it from "my" standard sphinx github action workflow. As a first step I just wanted to introduce sphinx and the "happy" path I already know, then migrate to the `sphinx-github-pages generator`. Once this is merged I'll planed to create a follow up, see https://github.com/exasol/sqlalchemy-exasol/issues/170#issue-1316667931.